File: py_app/util/encode/aes.py
import base64
from Crypto.Cipher import AES
from Crypto.Util.Padding import unpad, pad


class AesHelper:
    def __init__(self, key=None):
        # 默认密钥，如果没有传递则使用该默认密钥
        if key is None:
            key = b'ATOa4iA9Bv7jVdQis9#gnaWoaL&YuGae'  # 默认密钥
        self.key = key

    def decrypt(self, data):
        # 直接使用加密数据，不做截断
        # 创建解密对象，AES.new 会自动处理前 16 字节为 IV
        cipher = AES.new(self.key, AES.MODE_CBC, iv=self.key[:AES.block_size])
        # 解密整个数据
        decrypted = unpad(cipher.decrypt(data[0:]), AES.block_size)
        
        return decrypted

    def decrypt_and_base64(self, data):
        # 解码 Base64 数据并解密
        decoded_data = base64.b64decode(data)
        return self.decrypt(decoded_data)


def encrypt_to_base64(plaintext: str, secret_key: str) -> str:
    key_bytes = secret_key.encode("utf-8")
    plaintext_bytes = plaintext.encode("utf-8")
    padded_data = pad(plaintext_bytes, AES.block_size)
    cipher = AES.new(key_bytes, AES.MODE_ECB)
    ciphertext = cipher.encrypt(padded_data)
    return base64.b64encode(ciphertext).decode("utf-8")

File: ts_app/utils/encode/aes.ts
import * as crypto from 'crypto';

export class AesHelper {
  private key: Buffer;

  constructor(key?: string | Buffer) {
    // 默认密钥，如果未提供则使用默认
    if (!key) {
      key = 'ATOa4iA9Bv7jVdQis9#gnaWoaL&YuGae'; // 32 字节
    }
    this.key = typeof key === 'string' ? Buffer.from(key, 'utf-8') : key;
  }

  decrypt(data: Buffer): Buffer {
    const iv = this.key.slice(0, 16); // 使用 key 的前 16 字节作为 IV
    const decipher = crypto.createDecipheriv('aes-256-cbc', this.key, iv);
    decipher.setAutoPadding(false); // 手动处理 padding

    let decrypted = Buffer.concat([decipher.update(data), decipher.final()]);

    // 移除 PKCS#7 padding
    const padLength = decrypted[decrypted.length - 1];
    return decrypted.slice(0, decrypted.length - padLength);
  }

  decryptAndBase64(data: string): Buffer {
    const decoded = Buffer.from(data, 'base64');
    return this.decrypt(decoded);
  }
}
